salt/metalk8s/orchestrate/solutions/import-components.sls

```diff
@@ -0,0 +1,50 @@
+{%- set bootstrap_id = pillar.bootstrap_id %}
+{%- set version = pillar.metalk8s.nodes[bootstrap_id].version %}
+
+Import the Solutions archives:
+  salt.state:
+    - tgt: {{ bootstrap_id }}
+    - saltenv: metalk8s-{{ version }}
+    - sls:
+      - metalk8s.solutions.available
+
+Refresh master pillar:
+  module.run:
+    - saltutil.refresh_pillar: []
+    - require:
+      - salt: Import the Solutions archives
+
+Wait for master pillar to be refreshed:
+  module.run:
+    - test.sleep:
+      - length: 10
+    - require:
+      - module: Refresh master pillar
+
+Update Solutions configuration:
+  metalk8s_kubernetes.object_updated:
+    - name: metalk8s-solutions
+    - namespace: metalk8s-solutions
+    - kind: ConfigMap
+    - apiVersion: v1
+    - patch:
+        # here we can't use directly saltenv relative path with
+        # slsutil.renderer as it does not handle it in version 2018.3.4:
+        # https://github.com/saltstack/salt/pull/52293
+        __slot__:salt:slsutil.renderer({{
+          salt.cp.get_url(
+            "salt://" ~ slspath ~ "/files/configmap/content.jinja",
+            saltenv="metalk8s-" ~ version,
+          )
+        }}, jinja)
```

Comment:
Maybe cleaner to use a full manifest file with some jinja instead of only get a content for patch

```
Update Solutions configuration:
  metalk8s_kubernetes.object_updated:
    - name: {{ slspath }}/files/configmap/patch.yaml.jinja

Reply:
Tried this, but it does not seem to work, pillar is never up to date when rendering the template...

Reply:
Replaced by an orchestrate, it works like a charm.